pkg/porter/install.go

```diff
@@ -103,32 +103,35 @@ func (p *Porter) applyActionOptionsToInstallation(i *claims.Installation, opts *
 	if err != nil {
 		return err
 	}
-	if i.Parameters == nil {
-		i.Parameters = make(map[string]interface{}, len(opts.parsedParams))
-	}
 	// Record the user-specified parameter values
-	for k, v := range opts.parsedParams {
-		i.Parameters[k] = v
+	err = opts.populateInternalParameterSet(p, opts.bundleRef.Definition, i)
+	if err != nil {
+		return err
 	}
+
 	// Record the names of the parameter sets used
-	for _, ps := range opts.ParameterSets {
-		for _, existing := range i.ParameterSets {
-			if existing == ps {
-				continue
-			}
-		}
-		i.ParameterSets = append(i.ParameterSets, ps)
-	}
+	i.ParameterSets = append(i.ParameterSets, Unique(i.ParameterSets, opts.ParameterSets...)...)
 
 	// Record the names of the credential sets used
-	for _, cs := range opts.CredentialIdentifiers {
-		for _, existing := range i.ParameterSets {
-			if existing == cs {
-				continue
-			}
+	i.CredentialSets = append(i.CredentialSets, Unique(i.CredentialSets, opts.CredentialIdentifiers...)...)
+
+	return nil
+}
+
+func Unique(existings []string, n ...string) []string {
```

Comment:
I've marked this PR as fixing #1795. Please add a unit test for this function, and maybe another unit or smoke test that checks that calling say "upgrade" twice doesn't result in the names of the parameter sets having duplicates.

Reply:
I have added a unit test and a test case in the hello_test.go in smoke test